velero-csi-plugin/csi_snapshotter.go

```diff
@@ -45,55 +54,75 @@ type CSISnapshotter struct {
 func (p *CSISnapshotter) AppliesTo() (velero.ResourceSelector, error) {
 	p.log.Info("CSISnapshotter AppliesTo")
 
+	//TODO: Update this to look at only PVCs with "magic" label
```

Comment:
can you clarify what this means? I'm not clear on what the "magic" label is

Reply:
If the "magic" label has to do with restic selection, see this discussion on the CSI design proposal - it's tricky to get the ordering right :( https://github.com/vmware-tanzu/velero/pull/1661#discussion_r315375468

Reply:
I wasn't able to quickly come up w/ a way to describe that label. Hence the "magic". But the idea was to not have this plugin run for all PVCs, but just on the ones that have this label. I was not convinced we needed this label and hence the vague comment. 
I hadn't read that comment before. Based on what I read there, and the fact that there are checks we run to only operate on CSI PVs. I am convinced that the label is not required and will remove this comment also.

Reply:
Yeah, I agree that if we can not have this plugin run all the time that would be ideal. I think moving the CSI check up and exiting the plugin early if it's not a CSI volume is probably the best we can do for now. That way, we at least minimize the work the plugin does.

Reply:
marking this as resolved.